container/go/pkg/compat/overrideConfig.go

```diff
@@ -0,0 +1,545 @@
+	}
+	return expandedVar, nil
+}
+
+// getCreationTime returns the correct creation time for which to override the current config file.
+func getCreationTime(overrideInfo *OverrideConfigOpts) (time.Time, error) {
+	// createTime stores the input creation time with macros substituted.
+	var createTime string
+	var creationTime time.Time
+	var err error
+	// Parse for specific time formats.
+	if overrideInfo.CreationTimeString != "" {
+		var unixTime float64
+		// Use stamp to as preliminary replacement.
+		if createTime, err = Stamp(overrideInfo.CreationTimeString, overrideInfo.StampInfoFile); err != nil {
+			return time.Unix(0, 0), errors.Wrapf(err, "unable to format creation time from BUILD_TIMESTAMP macros")
+		}
+		// If creationTime is parsable as a floating point type, assume unix epoch timestamp.
+		// otherwise, assume RFC 3339 date/time format.
+		unixTime, err := strconv.ParseFloat(createTime, 64)
+		// Assume RFC 3339 date/time format. No err means it is parsable as floating point.
+		if err == nil {
+			// Ensure that the parsed time is within the floating point range.
+			// Values > 1e11 are assumed to be unix epoch milliseconds.
+			if unixTime > 1.0e+11 {
+				unixTime = unixTime / 1000.0
+			}
+			// Construct a RFC 3339 date/time from Unix epoch.
+			sec, dec := math.Modf(unixTime)
+			creationTime = time.Unix(int64(sec), int64(dec*(1e9))).UTC()
+		} else {
+			creationTime, err = time.Parse(time.RFC3339, createTime)
+			if err != nil {
+				return time.Unix(0, 0), errors.Wrapf(err, "failed to parse the %s into float, so assuming RFC3339", createTime)
+			}
+		}
+	}
+	return creationTime, nil
+}
+
+// updateConfigInfo returns an updated version of the input infoToStamp.
+func updateConfigInfo(overrideInfo *OverrideConfigOpts, infoToStamp []string) ([]string, error) {
+	var infoStamped = infoToStamp
+	for i, entry := range infoToStamp {
+		stampedEntry, err := Stamp(entry, overrideInfo.StampInfoFile)
+		if err != nil {
+			return []string{}, errors.Wrapf(err, "unable to perform substitutions to Env variable %s", entry)
+		}
+		infoStamped[i] = stampedEntry
+	}
+
+	return infoStamped, nil
+}
+
+// resolveEnvironment returns a string array with fully substituted environment variables.
+func resolveEnvironment(overrideInfo *OverrideConfigOpts, environMap map[string]string) ([]string, error) {
+	var baseEnvMap = make(map[string]string)
+	var err error
+	if len(overrideInfo.ConfigFile.Config.Env) > 0 {
+		baseEnvMap, err = keyValueToMap(overrideInfo.ConfigFile.Config.Env)
+		if err != nil {
+			return []string{}, errors.Wrapf(err, "error converting Config env array %v to map", overrideInfo.ConfigFile.Config.Env)
+		}
+	}
+
+	for k, v := range environMap {
+		var expanded string
+		if expanded, err = resolveVariables(v, baseEnvMap); err != nil {
+			return []string{}, errors.Wrapf(err, "unable to resolve environment variables in %s with content mapping at %s", k, v)
+		}
+		if _, ok := environMap[k]; ok {
```

Comment:
I don't think this condition can ever fail given the for loop is iterating over the keys of environMap